src/components/Single-Article.jsx

```diff
@@ -1,56 +1,80 @@
-import { useParams } from 'react-router-dom';
-import { useEffect, useState } from 'react';
-import { fetchArticleById, fetchCommentsForArticle } from '../utils/api';
-import styles from '../SingleArticle.module.css'
+import { useParams } from "react-router-dom";
+import { useEffect, useState } from "react";
+import {
+  fetchArticleById,
+  fetchCommentsForArticle,
+  sendVoteByArticleId,
+  removeVoteByArticleId
+} from "../utils/api";
+import styles from "../SingleArticle.module.css";
 
 const SingleArticle = (props) => {
-console.log(props)
-    const [article, setArticle] = useState([])
-    const [comments, setComments] = useState([])
-  const { article_id } = useParams(); 
+  const [article, setArticle] = useState([]);
+  const [comments, setComments] = useState([]);
+  const [votes, setVotes] = useState(0);
+  const [err, setErr] = useState(null);
 
+  const { article_id } = useParams();
 
   useEffect(() => {
     fetchArticleById(article_id).then((articleData) => {
-
       setArticle(articleData);
+      setVotes(articleData.votes);
     });
   }, [article_id]);
 
- 
-
   useEffect(() => {
     fetchCommentsForArticle(article_id).then((commentData) => {
-        
       setComments(commentData);
     });
   }, [article_id]);
 
+  function handleClick() {
+    setVotes((votes) => votes + 1);
+    setErr(null);
+    sendVoteByArticleId(article_id).catch((err) => {
+      setVotes((votes) => votes - 1);
+      setErr("Something went wrong ): please try again.");
+    });
+  }
+
```

Comment:
The function naming could be a bit better - handleClick is very generic. It would be good to mention how that click is an upvote, like in the function just below